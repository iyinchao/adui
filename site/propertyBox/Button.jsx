import React from "react"
import styles from "./components/propertyBox.scss"
import CodeBox from "./components/CodeBox"
import { Switch, Button } from "../../components"
import { SIZES, INTENTS } from "./consts"

export default class PropertyBox extends React.Component {
  state = {
    disabled: false,
    href: null,
    intent: "normal",
    leftIcon: "file-outlined",
    loading: false,
    rightIcon: "arrow-forward",
    size: "small",
    theme: null,
  }

  render() {
    const {
      disabled,
      href,
      intent,
      leftIcon,
      loading,
      rightIcon,
      size,
      theme,
    } = this.state

    const codes = `<Button
  ${disabled ? "disabled" : ""}
  ${href ? `href="${href}"` : ""}
  ${intent === "normal" ? "" : `intent="${intent}"`}
  ${leftIcon === null ? "" : `leftIcon="${leftIcon}"`}
  ${loading ? "loading" : ""}
  ${rightIcon === null ? "" : `rightIcon="${rightIcon}"`}
  ${size === "small" ? "" : `size="${size}"`}
  ${theme ? `theme="${theme}"` : ""}
>
  点击按钮
</Button>`

    return (
      <div className={styles.wrapper}>
        <div className={styles.property}>
          <div className={styles.left}>
            <Button {...this.state}>点击按钮</Button>
          </div>
          <div className={styles.right}>
            <div className={styles.rightTitle}>可配置项</div>
            <dl className={styles.controls}>
              <dt className={styles.controlTitle}>尺寸</dt>
              <dd className={styles.controlContent}>
                <Button.Group banner>
                  {SIZES.map(({ value, text }) => (
                    <Button
                      key={value}
                      active={value === size}
                      onClick={() => this.setState({ size: value })}
                    >
                      {text}
                    </Button>
                  ))}
                </Button.Group>
              </dd>
              <dt className={styles.controlTitle}>倾向</dt>
              <dd className={styles.controlContent}>
                <Button.Group banner>
                  {INTENTS.map(({ value, text }) => (
                    <Button
                      key={value}
                      active={value === intent}
                      onClick={() => this.setState({ intent: value })}
                    >
                      {text}
                    </Button>
                  ))}
                </Button.Group>
              </dd>
              <div className={styles.switchControl}>
                <Switch
                  checkedText="左图标"
                  unCheckedText="左图标"
                  checked={!!leftIcon}
                  onChange={(param) =>
                    this.setState({ leftIcon: param ? "file-outlined" : null })
                  }
                />
              </div>
              <div className={styles.switchControl}>
                <Switch
                  checkedText="右图标"
                  unCheckedText="右图标"
                  checked={!!rightIcon}
                  onChange={(param) =>
                    this.setState({ rightIcon: param ? "arrow-forward" : null })
                  }
                />
              </div>
              <div className={styles.switchControl}>
                <Switch
                  checkedText="加载态"
                  unCheckedText="加载态"
                  checked={loading}
                  onChange={(param) => this.setState({ loading: param })}
                />
              </div>
              <div className={styles.switchControl}>
                <Switch
                  checkedText="禁用态"
                  unCheckedText="禁用态"
                  checked={disabled}
                  onChange={(param) => this.setState({ disabled: param })}
                />
              </div>
              <div className={styles.switchControl}>
                <Switch
                  checkedText="轻量风格"
                  unCheckedText="轻量风格"
                  checked={!!theme}
                  onChange={(param) =>
                    this.setState({ theme: param ? "light" : null })
                  }
                />
              </div>
              <div className={styles.switchControl}>
                <Switch
                  checkedText="跳转链接"
                  unCheckedText="跳转链接"
                  checked={!!href}
                  onChange={(param) =>
                    this.setState({
                      href: param ? "https://ad.weixin.qq.com" : null,
                    })
                  }
                />
              </div>
            </dl>
          </div>
        </div>
        <CodeBox codes={codes} />
      </div>
    )
  }
}
