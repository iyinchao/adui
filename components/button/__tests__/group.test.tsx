import * as React from "react"
import TestRenderer from "react-test-renderer"
import Button from ".."

describe("Button.Group", () => {
  it("是否正确渲染", () => {
    const wrapper = TestRenderer.create(
      <Button.Group size="large">
        <Button>按钮一</Button>
        <Button>按钮二</Button>
      </Button.Group>
    )
    expect(wrapper).toMatchSnapshot()
  })
})
