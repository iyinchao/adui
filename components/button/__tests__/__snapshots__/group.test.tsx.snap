// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button.Group 是否正确渲染 1`] = `
<div
  className="adui-button-group"
>
  <button
    className="adui-button-base adui-button-normal adui-button-large"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    <span
      className="adui-button-content"
    >
      按钮一
    </span>
  </button>
  <button
    className="adui-button-base adui-button-normal adui-button-large"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    <span
      className="adui-button-content"
    >
      按钮二
    </span>
  </button>
</div>
`;
